packages/common/src/types.ts

```diff
@@ -118,20 +118,31 @@ type PropPanelProps = {
 
 export type PropPanelWidgetProps = _PropPanelWidgetProps & PropPanelProps;
 
+/**
+ * Type for the properties passed when the default schema is generated by a function.
+ * @property {UIOptions} options - Options object passed from the Viewer, Form, or Designer.
+ */
+export type DefaultSchemaProps = {
+  options: UIOptions;
+};
+
+
+
 /**
  * Used for customizing the property panel.
  * @template T - Type of the extended Schema object.
  * @property {Record<string, PropPanelSchema> | ((propPanelProps: Omit<PropPanelProps, 'rootElement'>) => Record<string, PropPanelSchema>)} schema - A function returning a form-render schema object or the schema object itself. When a function, it takes properties passed from the designer as arguments.
  * @property {Record<string, (props: PropPanelWidgetProps) => void>} [widgets] - An object of functions returning form-render widgets. The functions take, as arguments, both form-render's WidgetProps and properties passed from the designer.
  * @property {T} defaultSchema - The default schema set when adding the schema.
+ * @property {(props: DefaultSchemaProps) => T)} defaultSchemaFn - An optional function to generate the base schema dynamically using run-time args
  */
 export interface PropPanel<T extends Schema> {
   schema:
     | ((propPanelProps: Omit<PropPanelProps, 'rootElement'>) => Record<string, PropPanelSchema>)
     | Record<string, PropPanelSchema>;
-
   widgets?: Record<string, (props: PropPanelWidgetProps) => void>;
   defaultSchema: T;
+  defaultSchemaFn?: (props: DefaultSchemaProps) => T;
```

Comment:
I couldn't think of a better way of supporting a function than this...

Reply:
How about this?

```ts
  defaultSchema: 
  | T 
  | ((props: DefaultSchemaProps) => T);
```

Reply:
Yeah we could do that.

There are quite a lot of places in the code that use the `defaultSchema` that would need to be updated to check if it's a function though, not just the one place I changed in the LeftSidebar.

Reply:
i'll try 👍

Reply:
I tried this:

```
  if (schema.fontName == undefined) {
    onChange && onChange({ key: 'fontName', value: getFallbackFontName(font) });
  }
```

it works for text schema, but for MVT and tables it causes infinite loop.

I think the previous suggestion will be simpler than I thought, I will try it